### ISSUE TITLE
Loquibot does not work even under VxKex

### ISSUE THREAD

**Issue by HyperSoop** [vxiiduu/VxKex#38]
https://loquibot.com/

It seems from the logs that VxKex actually helps a bit, but not enough.  0.0.0.3.
It seems like it uses Python 3.10, which is not supported on Windows 7 - maybe that's the issue.
![image](https://user-images.githubusercontent.com/85982640/196002449-623c1d3b-5aa9-44a1-8ac1-8ce8a543d300.png)
![image1](https://user-images.githubusercontent.com/85982640/196002452-f5c03386-f82b-4cec-ac12-f76dc9e0169d.png)
![image2](https://user-images.githubusercontent.com/85982640/196002477-c92e5cbe-8c14-45e4-afa7-f6c9d07fab8c.png)

**Comment by okt04175** [comment]
Python 3.9.x 3.10.x For Win7 

Python 3.9 and 3.10 installers that support Windows 7 and Windows Server 2008 R2
https://github.com/adang1345/PythonWin7

**Comment by HyperSoop** [comment]
> Python 3.9.x 3.10.x For Win7
> 
> Python 3.9 and 3.10 installers that support Windows 7 and Windows Server 2008 R2 https://github.com/adang1345/PythonWin7

Thing is, it appears to use a bundled python310.dll. I'll install that anyway, maybe it'll help.

**Comment by HyperSoop** [comment]
![image](https://user-images.githubusercontent.com/85982640/196020539-34d78802-8b30-4035-8393-f27df774a715.png)
Sus. I'm better than this.

**Comment by HyperSoop** [comment]
Turns out the official installer does work under VxKex, but still no way to launch loquibot, because it only uses the bundled DLL (and there is no way to use vxkex on a dll, lol)

**Comment by dotexe1337** [comment]
> Turns out the official installer does work under VxKex, but still no way to launch loquibot, because it only uses the bundled DLL (and there is no way to use vxkex on a dll, lol)

if you are running the program with vxkex then it will inject into the python dll too.
but, current vxkex version is very buggy & unstable. vxiiduu is rewriting everything so i am waiting for that to be pushed to upstream then i will start working on my fork again.

i think in this case it is an issue with pyinstaller rather than python, if you look at the error it says there is a function missing.

> Sus. I'm better than this.

i think it is a false positive. only "detected" by kaspersky & zonealarm and those are not exactly regarded as being the best AVs (but really no AV is good, they are just bloat that wont detect the real nasty stuff anyways)

**Comment by vxiiduu** [comment]
> i am still too much of a noob to understand what a lot of this means (my whole point on working on my fork is to learn though), but it sounds like a big leap. i guess this will fix a lot of the issues with apps just crashing and having other random problems?

Yeah. Things are crashing because the code is just kind of crap at the moment. The reason for this is because the whole thing is based off some really quickly made proof of concept just to test whether this whole idea would work. Also, VxKex was pretty much my first programming project on Windows, I basically only made hello world and other simple stuff on linux before.

A lot of installers are just stub programs that unpack the real installer and run it, and then do something with its process handle. The way vxkex works right now is that programs can't directly start other programs, they have to go through vxkexldr first. This breaks a lot of apps apparently. A lot of installers are very badly coded and are sometimes harder to get working than the actual app.

Also the vxkexldr architecture makes it almost impossible to actually debug apps while vxkex is running. My new design is supposed to make it much easier to debug and monitor what the apps are doing, because otherwise it's like poking in the dark.

Anyway, the good news is that the data from my pc is probably fine. I just need a new motherboard because the old one somehow burned and isn't giving any power to the cpu.

**Comment by dotexe1337** [comment]
> Yeah. Things are crashing because the code is just kind of crap at the moment. The reason for this is because the whole thing is based off some really quickly made proof of concept just to test whether this whole idea would work. Also, VxKex was pretty much my first programming project on Windows, I basically only made hello world and other simple stuff on linux before.
> 
> A lot of installers are just stub programs that unpack the real installer and run it, and then do something with its process handle. The way vxkex works right now is that programs can't directly start other programs, they have to go through vxkexldr first. This breaks a lot of apps apparently. A lot of installers are very badly coded and are sometimes harder to get working than the actual app.
> 
> Also the vxkexldr architecture makes it almost impossible to actually debug apps while vxkex is running. My new design is supposed to make it much easier to debug and monitor what the apps are doing, because otherwise it's like poking in the dark.
> 
> Anyway, the good news is that the data from my pc is probably fine. I just need a new motherboard because the old one somehow burned and isn't giving any power to the cpu.

awesome, i cant wait for it to be done :D
